sound/soc/sof/pm.c

```diff
@@ -285,14 +324,10 @@ EXPORT_SYMBOL(snd_sof_resume);
 int snd_sof_suspend(struct device *dev)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
-	int ret;
-
-	if (snd_sof_dsp_d0i3_on_suspend(sdev))
-		return snd_sof_dsp_suspend(sdev, SOF_DSP_D0I3);
+	enum sof_dsp_power_state dsp_power_target;
 
-d3_suspend:
-	/* suspend to D3 */
-	return sof_suspend(dev, false);
+	dsp_power_target = snd_sof_dsp_power_target(sdev);
+	return sof_suspend(dev, false, dsp_power_target);
```

Comment:
so now this is unifying again the suspend routines in a single place.

Maybe this should the first cleanup really?